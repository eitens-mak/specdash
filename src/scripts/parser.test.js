import { createElement } from './parser.js';

it.each([
    [
	{
	    "title": "Sample Vega",
	    "type": "vega",
	    "spec": "public/line-chart.vg.json"
	}
    ],
    [
	{
	    "title": "Example JSON Editor",
	    "type": "json-editor",
	    "spec": "public/basic_person.json"
	}
    ]
])(
    "createElement",
    (ele) => {
	fetch.mockResponseOnce(
	    JSON.stringify(require('../../' + ele.spec))
	);

	return createElement(ele).then(card => {
	    expect(card.html()).toMatchSnapshot();
	});  
    },
)

it.each
