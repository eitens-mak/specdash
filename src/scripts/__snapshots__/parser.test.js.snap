// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createElement 1`] = `"<div class=\\"card-body\\"><h5 class=\\"card-title\\">Sample Vega</h5><div id=\\"specdashundefined\\"></div></div>"`;

exports[`createElement 2`] = `"<div class=\\"card-body\\"><h5 class=\\"card-title\\">Example JSON Editor</h5><div id=\\"specdashundefined\\"></div></div>"`;
